misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -183,4 +178,25 @@ private void createReport(final IBundleCoverage bundleCoverage, ModuleCoverage m
 
         }
     }
+
+    private List<Path> filterPaths(Path moduleJarPath, JarResolver jarResolver) {
+        List<Path> filteredPathList = new ArrayList<>();
+        List<Path> pathCollection;
+
+        try {
+            pathCollection =
+                    new ArrayList<>(jarResolver.getJarFilePathsRequiredForTestExecution(this.module.moduleName()));
+        } catch (IllegalStateException e) {
+            pathCollection = new ArrayList<>(jarResolver.getJarFilePathsRequiredForExecution());
```

Comment:
Why do we catch and add the `getJarFilePathsRequiredForExecution`? The  `getJarFilePathsRequiredForTestExecution` is supposed to have all the required jars. So I think we should simply throw the exception.

Reply:
In this case, `getJarFilePathsRequiredForTestExecution` would throw an IllegalStateException when it fails to find the relevant `-testable` jar for modules without tests. 

By catching the exception, and calling `getJarFilePathRequiredForExecution` I'm making sure that for those modules without tests, the default jars are being collected properly and passed to jacoco. 

the pathCollection cannot be left null either, since the entire code coverage would stop executing as a result of `jars not found`.

Reply:
In that case, instead of catching the IllegalStateException we can write a condition to check if the module contains tests. `module.testDocumentIds().isEmpty()` can be used for this

Reply:
Refactored to make use of module.testDocuementIds().isEmpty()